### PR TITLE
[GAL-5439] add default state for web search results 

### PR DESCRIPTION
**Pull request by Rohan-cp**
### Summary of Changes

Provide a brief description of the changes made in this PR. Explain the problem you are fixing or the feature you are adding.

### Demo 

| Before                                     | After                                      |
| ------------------------------------------ | ------------------------------------------ |
| Screenshot 2024-04-02 at 20 10 09 | Screenshot 2024-04-05 at 06 07 02 |

https://github.com/gallery-so/gallery/assets/49758803/15011f37-42f4-4404-b295-011cfa4c4977

### Moweb
Screenshot 2024-04-06 at 04 46 05

### Testing Steps

Provide steps on how the reviewer can test the changes.

### Checklist

Please make sure to review and check all of the following:

- [x] I've tested the changes and all tests pass.
- [x] (if web) I've tested the changes on various desktop screen sizes to ensure responsiveness.
- [ ] (if mobile) I've tested the changes on both light and dark modes.

### CODE REVIEW

File: apps/web/src/components/Feed/SuggestedProfileCard.tsx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+
+  const loggedInUserId = useLoggedInUserId(query);
+  const isOwnProfile = loggedInUserId && loggedInUserId === user?.id;
+
+  if (!user) {
+    throw new Error('No user available to showcase SuggestedProfileCard');
+  }
+
+  const { bio } = user;
+  const unescapedBio = useMemo(() => (bio ? unescape(bio) : ''), [bio]);
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/SuggestedProfileCard.tsx:76]: we shouldn't need to manually `unescape` on the frontend anymore, the backend should handle this!

File: apps/web/src/components/Feed/SuggestedProfileCard.tsx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+  const bioFirstLine = useMemo(() => {
+    if (!unescapedBio) {
+      return '';
+    }
+    return unescapedBio.split('\n')[0] ?? '';
+  }, [unescapedBio]);
```
Comment (Robinnnnn) [inline apps/web/src/components/Feed/SuggestedProfileCard.tsx:83]: should be able to shortcut the above few lines:

```js
user.bio?.split('\n')[0] ?? ''
```

File: apps/web/src/components/Feed/SuggestedProfileCard.tsx (hunk 1 above)
Comment (Robinnnnn) [reply apps/web/src/components/Feed/SuggestedProfileCard.tsx:76]: update: the backend unescaping happens in `schema.resolvers.helpers.go`: https://github.com/gallery-so/go-gallery/pull/1214/files

ask: in a follow-up PR, can you track down all the cases where the frontend calls `unescape` (screenshot), then ensure the necessary locations in the backend unescapes those fields? the end result should be that we don't call `unescape` anywhere in the frontend

image

File: apps/web/src/components/Feed/SuggestedProfileCard.tsx (hunk 2 above)
Comment (Rohan-cp) [reply apps/web/src/components/Feed/SuggestedProfileCard.tsx]: nice! made the change and checked it doesn't break anything

Screenshot 2024-04-08 at 18 33 48

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**